plans/restore.pp

```diff
@@ -0,0 +1,323 @@
+      )
+
+      run_task('peadm::restore_classification', $primary_target,
+        classification_file => "${recovery_directory}/transformed_classification.json",
+      )
+    } else {
+      run_task('peadm::restore_classification', $primary_target,
+        classification_file => "${recovery_directory}/classifier/classification_backup.json",
+      )
+    }
+  }
+
+  if $restore_type == 'recovery' {
+    out::message('# Restoring ca, certs, code and config for recovery')
+  # lint:ignore:strict_indent
+      run_command(@("CMD"/L), $primary_target)
+        /opt/puppetlabs/bin/puppet-backup restore \
+          --scope=certs,code,config \
+          --tempdir=${shellquote($recovery_directory)} \
+          --force \
+          ${shellquote($recovery_directory)}/recovery/pe_backup-*tgz
+        | CMD
+  # lint:endignore
+  } elsif $restore_type == 'recovery-db' {
+    out::message('# Restoring primary database for recovery')
+  } else {
+    if getvar('recovery_opts.ca') {
+      out::message('# Restoring ca and ssl certificates')
+  # lint:ignore:strict_indent
+      run_command(@("CMD"/L), $primary_target)
+        /opt/puppetlabs/bin/puppet-backup restore \
+          --scope=certs \
+          --tempdir=${shellquote($recovery_directory)} \
+          --force \
+          ${shellquote($recovery_directory)}/ca/pe_backup-*tgz
+        | CMD
+    }
+
+    if getvar('recovery_opts.code') {
+      out::message('# Restoring code')
+      run_command(@("CMD"/L), $primary_target)
+        /opt/puppetlabs/bin/puppet-backup restore \
+          --scope=code \
+          --tempdir=${shellquote($recovery_directory)} \
+          --force \
+          ${shellquote($recovery_directory)}/code/pe_backup-*tgz
+        | CMD
+    }
+
+    if getvar('recovery_opts.config') {
+      out::message('# Restoring config')
+      run_command(@("CMD"/L), $primary_target)
+        /opt/puppetlabs/bin/puppet-backup restore \
+          --scope=config \
+          --tempdir=${shellquote($recovery_directory)} \
+          --force \
+          ${shellquote($recovery_directory)}/config/pe_backup-*tgz
+        | CMD
+    }
+  }
+  # Use PuppetDB's /pdb/admin/v1/archive API to SAVE data currently in PuppetDB.
+  # Otherwise we'll completely lose it if/when we restore.
+  # TODO: consider adding a heuristic to skip when innappropriate due to size
+  #       or other factors.
+  if getvar('recovery_opts.puppetdb') and $restore_type == 'migration' {
+    out::message('# Exporting puppetdb')
+    run_command(@("CMD"/L), $primary_target)
+      /opt/puppetlabs/bin/puppet-db export \
+        --cert=$(/opt/puppetlabs/bin/puppet config print hostcert) \
+        --key=$(/opt/puppetlabs/bin/puppet config print hostprivkey) \
+        ${shellquote($recovery_directory)}/puppetdb-archive.bin
+      | CMD
+  }
+
+  ## shutdown services
+  run_command(@("CMD"/L), $primary_target)
+    systemctl stop pe-console-services pe-nginx pxp-agent pe-puppetserver \
```

Comment:
Is this supposed to stop all services? Because PE provides more.

Reply:
This is stopping all the services that we require for the time being.